packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -373,14 +373,15 @@ abstract class RenderSliverPinnedPersistentHeader extends RenderSliverPersistent
     final bool overlapsContent = constraints.overlap > 0.0;
     excludeFromSemanticsScrolling = overlapsContent || (constraints.scrollOffset > maxExtent - minExtent);
     layoutChild(constraints.scrollOffset, maxExtent, overlapsContent: overlapsContent);
-    final double layoutExtent = (maxExtent - constraints.scrollOffset).clamp(0.0, constraints.remainingPaintExtent) as double;
+    final double effectiveRemainingPaintExtent = math.max(0, constraints.remainingPaintExtent - constraints.overlap);
```

Comment:
Why do we need to reduce the paintExtent by the overlap?

AFAIK you can paint into the area that's overlapped by a previous sliver.

Reply:
For future reference: because we set the paintOrigin below to overlap.